src/main/java/com/aws/greengrass/certificatemanager/CertificateManager.java

```diff
@@ -115,9 +115,12 @@ public String getCaPassPhrase() {
      * @throws KeyStoreException if unable to access KeyStore
      * @throws CsrProcessingException if unable to process CSR
      */
+    // the service certificate subscribe function is invoked on the thread starting broker, i.e. nucleus
+    // lifecycle management thread
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public void subscribeToServerCertificateUpdates(@NonNull String csr, @NonNull Consumer<X509Certificate> cb)
```

Comment:
Can we also add a comment on where the callback is invoked?

Reply:
sure

Reply:
I don't think we should make a statement about the calling context here. It's another component calling this. This likely won't hold true for bring your own broker.

We should note in the method documentation itself that this will result in a certificate being generated and the callback will be invoked prior to this method returning. Issuing a certificate is computationally expensive, so the caller needs to make sure they call in an appropriate thread context.

I'd prefer that we document callback thread context in the code where we actually invoke the callback